frb_example/pure_dart_multi/dart/lib/test_api_block_2.dart

```diff
@@ -0,0 +1,141 @@
+import 'dart:typed_data';
```

Comment:
I guess you mean that there are many tests among the 3 regular Api blocks that serve similar test purposes.    

When rewriting the whole test example in `pure_multi`, I finally decided to take only 3 regular Api blocks, since this number is enough to cover all situations(hopefully I don't miss something), like some types are shared but not shared overall regular blocks.  

And for the tests inside each regular block, since `pure_multi` is a project treated as a combination of unit tests, I don't think it a good practice to "extract the common part"---just leave them decoupled is best for unit test.  Besides, though the test cases in each regular block look the same/symmetric from one another, actually they are just similar but not the same/symmetric if you check them one by one.

From the `main.dart`:
```dart
  testTypeMethods(api1, apiShared);
  testApiBlock1(api1, apiShared, enumList);
  testApiBlock2(api2, apiShared, enumList);
  testApiBlock3(api3, apiShared, enumList);
  testSharedObjOverBlocks(api1, api2, api3, apiShared, enumList);
```
Besides, it is easy to see there are 5 blocks of unit tests.  The only coupled block is `testSharedObjOverBlocks` because the tests there are used to test whether a dart object could be successfully transferred through different (regular) blocks.

So to sum up, I would still prefer to leave it simple (though a little heavy code) and decoupled if possible, just like what you mentioned when I wrote complex but short unit tests in Rust from early pr.